src/python/pants/backend/python/tasks/python_repl.py

```diff
@@ -30,6 +31,8 @@ def register_options(cls, register):
              help='The IPython REPL entry point.')
     register('--ipython-requirements', advanced=True, type=list, default=['ipython==1.0.0'],
              help='The IPython interpreter version to use.')
+    register('--interpreter-compatibility', advanced=True, type=list, default=[],
```

Comment:
After exploring options further, I noticed that these are namespaced to `--repl-py`, so we might be able to shorten this to just `--compatibility`. Keeping these options short and sweet makes them easier to use for the end developer. The conceptual model should carry over from the target-level compatibility settings, but I would like some input from other Pants contributors before we change it.